CMakeLists.txt

```diff
@@ -151,6 +151,10 @@ cuda_include_directories(
   "${PROJECT_SOURCE_DIR}"
   "${PROJECT_SOURCE_DIR}/include")
 
+
+set(CHECK_TARGET "check")
```

Comment:
Out of curiosity: why not "test"?

Reply:
I got inspired by https://clang.llvm.org/get_started.html where they have make check-clang.
I didn't want to call it `test`, because `make test` could also suggest the building of tests (which we control by `BUILD_TEST` CMake flag).